src/Controller/CRUDController.php

```diff
@@ -779,11 +798,10 @@ public function historyAction($id = null)
      *
      * @return Response
      */
-    public function historyViewRevisionAction($id = null, $revision = null)
+    public function historyViewRevisionAction($id = null, $revision = null) // NEXT_MAJOR: Remove the unused $id parameter
     {
         $request = $this->getRequest();
```

Comment:
```suggestion
        if (null !== $id) {
            @trigger_error(
                sprintf('Support for the "id" route param as argument 1 at `%s()` is deprecated since sonata-project/admin-bundle 3.x and will be removed in 4.0, use `AdminInterface::getIdParameter()` instead.', __METHOD__),
                E_USER_DEPRECATED
            );
        }

        $request = $this->getRequest();
```

Reply:
I don't see the point in this.

As said before, the id is in the route, so it won't be null and I'll get deprecations everytime.
And even if I rename $id to $deprecatedId like the others,
The way to fix the deprecation is by calling `historyViewRevisionAction(null, $revision)` which means it still needed to update the code after the major update.

The better is the enemy of the good.
This PR is opened for two months, just for deprecating a useless parameter. I really would prefer to focus my time on working on the Symfony 5 support and the Sonata major release.

Reply:
Leave then the decision to the rest of the team :+1:
Thank you so much for the time spent working with this PR and its comments.

Reply:
To me, it looks good as is.

Reply:
No problem.
50% of our team-project is based on Sonata, it's normal to contribute.